server/src/main/java/com/linecorp/centraldogma/server/internal/api/TokenService.java

```diff
@@ -0,0 +1,156 @@
+import com.linecorp.centraldogma.server.internal.metadata.HolderWithLocation;
+import com.linecorp.centraldogma.server.internal.metadata.MetadataService;
+import com.linecorp.centraldogma.server.internal.metadata.Token;
+import com.linecorp.centraldogma.server.internal.metadata.Tokens;
+import com.linecorp.centraldogma.server.internal.storage.project.ProjectManager;
+
+/**
+ * Annotated service object for managing {@link Token}s.
+ */
+@ExceptionHandler(HttpApiExceptionHandler.class)
+public class TokenService extends AbstractService {
+    private static final Logger logger = LoggerFactory.getLogger(TokenService.class);
+
+    private static final JsonNode activation = Jackson.valueToTree(
+            ImmutableList.of(
+                    ImmutableMap.of("op", "replace",
+                                    "path", "/status",
+                                    "value", "active")));
+    private static final JsonNode deactivation = Jackson.valueToTree(
+            ImmutableList.of(
+                    ImmutableMap.of("op", "replace",
+                                    "path", "/status",
+                                    "value", "inactive")));
+
+    private final MetadataService mds;
+
+    public TokenService(ProjectManager projectManager, CommandExecutor executor,
+                        MetadataService mds) {
+        super(projectManager, executor);
+        this.mds = requireNonNull(mds, "mds");
+    }
+
+    /**
+     * GET /tokens
+     *
+     * <p>Returns the list of the tokens generated before.
+     */
+    @Get("/tokens")
+    public CompletionStage<Collection<Token>> listTokens() {
+        return mds.getTokens()
+                  .thenApply(Tokens::withoutSecret)
+                  .thenApply(tokens -> tokens.appIds().values());
+    }
+
+    /**
+     * POST /tokens
+     *
+     * <p>Returns a newly-generated token belonging to the current login user.
+     */
+    @Post("/tokens")
+    @ResponseConverter(CreateApiResponseConverter.class)
+    public CompletionStage<HolderWithLocation<Token>> createToken(@Param("appId") String appId,
+                                                                  @RequestObject Author author) {
+        return mds.createToken(author, appId)
+                  .thenCompose(unused -> mds.findTokenByAppId(appId))
+                  .thenApply(token -> HolderWithLocation.of(token, "/tokens/" + appId));
+    }
+
+    /**
+     * DELETE /tokens/{appId}
+     *
+     * <p>Deletes a token of the specified ID then returns it.
+     */
+    @Delete("/tokens/{appId}")
+    public CompletionStage<Token> deleteToken(@Param("appId") String appId,
+                                              @RequestObject Author author,
+                                              @RequestObject User loginUser) {
+        return getTokenOrRespondForbidden(appId, loginUser).thenCompose(
+                token -> mds.destroyToken(author, appId)
```

Comment:
just wondering is there any reason that you used `destroy` instead of `delete`?

Reply:
`removeToken()` removes a token from a project. And `destroyToken()` removes a token from the system completely (removing it from every project and the system). So I want to make its meaning clear by using `destroy` instead of `delete`.